src/codewhisperer/util/supplementalContext/codeParsingUtil.ts

```diff
@@ -68,29 +67,21 @@ export function countSubstringMatches(arr1: string[], arr2: string[]): number {
     return count
 }
 
-export async function isTestFile(editor: vscode.TextEditor, dependencyGraph: DependencyGraph): Promise<boolean> {
-    const languageConfig = utgLanguageConfigs[editor.document.languageId]
-    if (!languageConfig) {
-        // We have enabled the support only for python and Java for this check
-        // as we depend on Regex for this validation.
-        return false
-    }
+export async function isTestFile(
+    filePath: string,
+    languageConfig: { languageId: string; dependencyGraph?: DependencyGraph; fileContent?: string }
+): Promise<boolean> {
+    const byPath = filePath.includes(`tests/`) || filePath.includes('test/') || filePath.includes('tst/')
+    const byName = isTestFileByName(filePath, languageConfig.languageId)
+    const byDependencyGraph =
```

Comment:
If `await languageConfig.dependencyGraph.isTestFile(languageConfig.fileContent)` is slow, you can consider 

```
if ( byPath || by Name) return true.
return byDependencyGraph
```

Consider rename byPath --> filePathContainsTest, byName --> fileNameFitsTestPattern, 
byDependencyGraph --> 

To make this code more readable

Reply:
updated